src/hooks/Tokens.ts

```diff
@@ -28,8 +29,36 @@ function useTokensFromMap(tokenMap: TokenAddressMap, chainId: Maybe<SupportedCha
   }, [chainId, tokenMap])
 }
 
-export function useAllTokensMultichain(): TokenAddressMap {
-  return useCombinedTokenMapFromUrls(DEFAULT_LIST_OF_LISTS)
+export type ChainTokenMap = { [chainId in number]?: { [address in string]?: Token } }
+export function useAllTokensMultichain(): ChainTokenMap {
+  const allTokensFromLists = useCombinedTokenMapFromUrls(DEFAULT_LIST_OF_LISTS)
+  const userAddedTokensMap = useAppSelector(({ user: { tokens } }) => tokens)
+
+  return useMemo(() => {
+    const chainTokenMap: ChainTokenMap = {}
+
+    Object.keys(userAddedTokensMap)
+      .map(Number)
```

Comment:
maybe you should cast these keys to `Number` in the forEach body itself, instead of `map` - then we would iterate through the array once instead of twice, right?